arrayfire/library.py

```diff
@@ -644,7 +655,7 @@ def __init__(self):
 
         # Iterate in reverse order of preference
         for name in ('cpu', 'opencl', 'cuda', ''):
-            libnames = self.__libname(name)
+            libnames = reversed(self.__libname(name))
```

Comment:
Why reverse the names ? Or is there something else happening down the line ?

Wouldn't the error messages that print these names show revered strings also ?

Reply:
it is a list of names we are reversing, one is the full path name to the library, the second is the short library only name that will search the default library paths. By reversing the order we make the local full path the preferred path. If it fails due to no local binaries it will use the second system path. This can otherwise be done in the __libname function, not sure where more appropriate

Reply:
Seems like I am misunderstanding the `reversed` call. I was thinking it is reversing the characters of string.